File: mobile/app/src/main/java/com/smartsuschef/mobile/network/dto/ForecastDtos.kt
package com.smartsuschef.mobile.network.dto

import com.google.gson.annotations.SerializedName

/**
 * Forecast DTO
 * Maps to: ForecastDto(string Date, string RecipeId, string RecipeName, int Quantity, List<ForecastIngredientDto> Ingredients)
 */
data class ForecastDto(
    @SerializedName("date")
    val date: String,

    @SerializedName("recipeId")
    val recipeId: String,

    @SerializedName("recipeName")
    val recipeName: String,

    @SerializedName("quantity")
    val quantity: Int,

    @SerializedName("confidence")
    val confidence: String,

    @SerializedName("ingredients")
    val ingredients: List<ForecastIngredientDto>
)

/**
 * Forecast Ingredient DTO
 * Maps to: ForecastIngredientDto
 */
data class ForecastIngredientDto(
    @SerializedName("ingredientId")
    val ingredientId: String,

    @SerializedName("ingredientName")
    val ingredientName: String,

    @SerializedName("unit")
    val unit: String,

    @SerializedName("quantity")
    val quantity: Double // decimal in C#
)

/**
 * Forecast Summary DTO
 * Maps to: ForecastSummaryDto(string Date, int TotalQuantity, decimal ChangePercentage)
 */
data class ForecastSummaryDto(
    @SerializedName("date")
    val date: String,

    @SerializedName("totalQuantity")
    val totalQuantity: Int,

    @SerializedName("changePercentage")
    val changePercentage: Double // decimal in C#
)

/**
 * Weather DTO
 * Maps to: WeatherDto
 */
data class WeatherDto(
    @SerializedName("temperature")
    val temperature: Double, // decimal in C#

    @SerializedName("condition")
    val condition: String,

    @SerializedName("humidity")
    val humidity: Int,

    @SerializedName("description")
    val description: String
)

/**
 * Holiday DTO
 * Maps to: HolidayDto
 */
data class HolidayDto(
    @SerializedName("date")
    val date: String,

    @SerializedName("name")
    val name: String
)

/**
 * Weather Forecast DTO
 * Maps to: WeatherForecastDto
 */
data class WeatherForecastDto(
    @SerializedName("date")
    val date: String,

    @SerializedName("temperatureMax")
    val temperatureMax: Double? = null,

    @SerializedName("temperatureMin")
    val temperatureMin: Double? = null,

    @SerializedName("rainMm")
    val rainMm: Double, // decimal in C#

    @SerializedName("weatherCode")
    val weatherCode: Int,

    @SerializedName("weatherDescription")
    val weatherDescription: String
)

/**
 * Calendar Day DTO
 * Maps to: CalendarDayDto - includes weather and holiday status
 */
data class CalendarDayDto(
    @SerializedName("date")
    val date: String,

    @SerializedName("isHoliday")
    val isHoliday: Boolean,

    @SerializedName("holidayName")
    val holidayName: String? = null,

    @SerializedName("isSchoolHoliday")
    val isSchoolHoliday: Boolean,

    @SerializedName("isWeekend")
    val isWeekend: Boolean,

    @SerializedName("weather")
    val weather: WeatherForecastDto? = null
)

/**
 * Tomorrow Forecast DTO
 * Maps to: TomorrowForecastDto - combined forecast and calendar response
 */
data class TomorrowForecastDto(
    @SerializedName("date")
    val date: String,

    @SerializedName("calendar")
    val calendar: CalendarDayDto,

    @SerializedName("weather")
    val weather: WeatherForecastDto? = null
)

File: backend/SmartSusChef.Api/DTOs/ForecastDtos.cs
namespace SmartSusChef.Api.DTOs;

public record ForecastDto(
    string Date,
    string RecipeId,
    string RecipeName,
    int Quantity,
    List<ForecastIngredientDto> Ingredients,
    string Confidence = "Medium" // Added Confidence field
);

public record ForecastIngredientDto(
    string IngredientId,
    string IngredientName,
    string Unit,
    decimal Quantity
);

public record ForecastSummaryDto(
    string Date,
    int TotalQuantity,
    decimal ChangePercentage
);

public record WeatherDto(
    decimal Temperature,
    string Condition,
    int Humidity,
    string Description
);

public record HolidayDto(
    string Date,
    string Name
);

/// <summary>
/// Weather forecast for a specific date
/// </summary>
public record WeatherForecastDto(
    string Date,
    decimal? TemperatureMax,
    decimal? TemperatureMin,
    decimal RainMm,
    int WeatherCode,
    string WeatherDescription
);

/// <summary>
/// Calendar info for a specific date including weather and holiday status
/// </summary>
public record CalendarDayDto(
    string Date,
    bool IsHoliday,
    string? HolidayName,
    bool IsSchoolHoliday,
    bool IsWeekend,
    WeatherForecastDto? Weather
);

/// <summary>
/// Combined forecast and calendar response for tomorrow
/// </summary>
public record TomorrowForecastDto(
    string Date,
    CalendarDayDto Calendar,
    WeatherForecastDto? Weather
);
